Lib/tempfile.py

```diff
@@ -98,7 +98,7 @@ def _infer_return_type(*args):
     """Look at the type of all args and divine their implied return type."""
     return_type = None
     for arg in args:
-        if arg is None:
+        if arg is None or isinstance(arg, _os.PathLike):
```

Comment:
I think `dir = _os.fspath(dir)` should be called before calling `_infer_return_type()`. It would be not correct to call `_os.fspath()` for non-path arguments.